astroid/nodes/scoped_nodes/scoped_nodes.py

```diff
@@ -133,24 +133,24 @@ def clean_typing_generic_mro(sequences: list[list[ClassDef]]) -> None:
     bases_mro.pop(position_in_inferred_bases)
 
 
-def clean_duplicates_mro(sequences, cls, context):
+def clean_duplicates_mro(
+    sequences: Iterable[Iterable[ClassDef]],
```

Comment:
https://github.com/pylint-dev/astroid/blob/feafcb83cf5b379cdb7613f48268a7525a591f40/astroid/nodes/scoped_nodes/scoped_nodes.py#L2806

This makes this `SuccesfulInferenceResult` although I wonder if `ClassDef.bases` could be narrowed down. I'd be okay with making this `ClassDef` for now and fixing later if necessary, but note that this will add one more `mypy` warning after `_inferred_bases` gets its signature typed.

Reply:
Yep. However, Pyright also says it could be `ClassDef` due to this:

https://github.com/pylint-dev/astroid/blob/feafcb83cf5b379cdb7613f48268a7525a591f40/astroid/nodes/scoped_nodes/scoped_nodes.py#L2803-L2804

Let's wait until we're ready to type this all together.

Reply:
Yeah of course, the `bases` line just broadens the type (perhaps/probably incorrectly)